### ISSUE TITLE
Make commit message editing work like normal git

### ISSUE THREAD

**Issue by aristidesfl** [divmain/GitSavvy#248]
The problem here is going back and forth between normal git and gitsavvy. The way the commit works is different, causing confusion.

In normal git you edit the commit message in your editor and close the editing window/buffer to submit the commit. To cancel the commit, just just provide an empty commit message.

In git-savvy closing the window cancels the commit,  and you need to press super+return in order to confirm the message and submit the commit.

GitSavvy way seems better to me, however, since there is no way to make normal git to behave like GitSavvy, it would be great to have an option to make GitSavvy behave like normal git when editing commit messages, for the sake of consistency.

**Comment by divmain** [comment]
I don't see resounding value in bringing the exact behavior to parity.  However, there might be merit in notifying someone when they try to do a commit with an empty message.  That would protect against a mistake when someone expects an empty commit message to result in no commit at all.  Let me think a bit about this!

**Comment by aristidesfl** [comment]
Notifying someone solves a problem, but not the main problem. Some people need to go back and forth between GitSavvy and normal git. If you don't do that, then you won't see the value. But someone who does will have to interrupt the process and stop to think if normal git or gitsavvy is being used and adjust behaviour accordingly. And mistakes will be made, and pain will be had and ultimately slow the process down.

**Comment by divmain** [comment]
Fair enough.  That last sentence sounds like a personal statement, although I hope something like this won't immediately turn you off to considering GitSavvy.

Maybe you can help me work through some of the UX, and that'll help me come to a conclusion on this.  Say we add a configuration option `commit_on_close`.  When `true`, closing a commit message window would result in a commit being done, except when the message is empty.  We could also update the helper text in the commit message window so there's no confusion.

However, we use the same machinery for things like amending a commit.  The mechanism for aborting an amend with `git` command line tools is to delete all of the commit message text.  That is, in my opinion, horrible UX, and things like this were the original inspiration for GitSavvy - to give Git a friendlier but still robust alternate interface in my every-day text editor.

I think it would be more inconsistent not to adopt this alternate behavior for amends.  But making that change seems like I'm adding a huge wart to the interface.  Can you think of any alternatives?

I will also have to do a review to see whether this adds significant complexity to the commit message implementation.  If it is simple enough, I may be fine with adding the option and just calling it out as "not recommended".

**Comment by aristidesfl** [comment]
I can't think of an alternative. The amend behaviour should also work like original git when this option is enabled, that is the point of the option after all, even though it might seem like a step backwards in term of usability. Only people who have used git for long enough will enable this option and they will know how to cancel a commit amend. Also there can be instructions on how to do it, when the option is enabled.

**Comment by sentience** [comment]
For what it’s worth, my first few weeks with GitSavvy were plagued by this issue. I was coming from SublimeGit, which works as described for “standard Git” (commit on closing the commit message buffer), so I was constantly writing a commit message and then closing the buffer, losing my writing. :(

Having adapted to GitSavvy’s ways, I am pretty comfortable with ⌘Enter (in fact, I now hit that when creating commits with command-line Git and am surprised when nothing happens), but I do see the argument for sticking with the standard. It would certainly have made adopting GitSavvy less painful for me.

As for the Amend Commit workflow, maybe it’s just me but backing out of an Amend just isn’t something I do often. Sublime Git had no advertised way to cancel an amend once you’d opened the buffer to edit the commit message. If I had to do it, I’d go to the Reflog (a pretty advanced/obscure feature, admittedly), but in the many months I used Sublime Git, I never had occasion to do it, and I amend a lot. (I just checked, @divmain, and you’re right—deleting the commit message entirely does cancel the amend in Sublime Git.)

To the extent that it’s practical, I’m happy with the proposal to make the behaviour an option that advanced users who want the “standard” workflow can toggle on. For my part, I probably would toggle it on and re-adapt.
